torchgen/model.py

```diff
@@ -847,9 +896,94 @@ def from_dict(
         return NativeFunctionsGroup(
             functional=functional,
             inplace=inplace,
+            mutable=mutable,
             out=out,
         )
 
+    # This function, given a "base" function of one SchemaKind, as well as a target SchemaKind,
+    # generates a new NativeFunction with the same properties, but using the target SchemaKind.
+    # We only actually generate functions for either functional or out= SchemaKinds.
+    @staticmethod
+    def generate_function(
+        f: NativeFunction, k: SchemaKind
+    ) -> Tuple[
+        NativeFunction, Dict[DispatchKey, Dict["OperatorName", "BackendMetadata"]]
```

Comment:
A comment explaining what the second element of the tuple is would be helpful. In particular, when you "generate" a native function this way, it also gives an obligation to do some code generation to actually produce the kernel for the newly generated kernel. To me, this suggests that this function shouldn't live on model, and it should live closer to the site where the generated code is actually being created.

Reply:
That being said, I don't practice what I preach; for ufuncs I was willing to goop up model.py with a forward declaration in the dispatch table for the ufunc generated kernel. 

```
        # Program the BackendIndex for the implicit dispatch entry from ufunc
        if ufunc_inner_loop:
            assert structured, "ufunc must be structured"
            for dispatch_key in STRUCTURED_DISPATCH_KEYS:
                assert (
                    dispatch_key not in dispatch
                ), f"ufunc should not have explicit dispatch entry for {dispatch_key}"
                dispatch[dispatch_key] = BackendMetadata(
                    kernel=ufunc.schema_kernel_name(func, dispatch_key), structured=True
                )
```

It's a little different though, I guess, because you can't do the additions on the fly; it has to be done as a separate postprocessing step. Still, I think the logic here should probably go with the postprocessing, since in general it's not sound to manufacture new native functions this way; this should be a controlled process that happens only at one place in the codegen.

Reply:
ok, moving this function into `gen.py` to live more directly with its (single) call site seems super reasonable.